components/mediation/inbound-endpoints/org.wso2.micro.integrator.inbound.endpoint/src/main/java/org/wso2/carbon/inbound/endpoint/internal/http/api/ConfigurationLoader.java

```diff
@@ -199,17 +212,19 @@ private static void populateHandlers(OMElement apiElement, InternalAPI api) {
         api.setHandlers(handlerList);
     }
 
-    private static InternalAPIHandler createHandler(String classFQName) {
+    private static InternalAPIHandler createHandler(String classFQName, String context, List<String> resources) {
 
         try {
-            Object obj = Class.forName(classFQName).newInstance();
+            Constructor c = Class.forName(classFQName).getConstructor(String.class);
+            Object obj = c.newInstance(context);
             if (obj instanceof InternalAPIHandler) {
+                ((InternalAPIHandler) obj).setResources(resources);
                 return (InternalAPIHandler) obj;
             } else {
                 throw new SynapseException("Error creating Internal InternalAPIHandler. "
                                                    + "The InternalAPIHandler should be of type InternalAPIHandler");
             }
-        } catch (ClassNotFoundException | InstantiationException | IllegalAccessException e) {
+        } catch (ClassNotFoundException | InstantiationException | IllegalAccessException | NoSuchMethodException | InvocationTargetException e) {
```

Comment:
Shall we format this as this exceeds 120 charecters.

Reply:
done